### ISSUE TITLE
PicklingError on flag.users

### ISSUE THREAD

**Issue by ryanisnan** [django-waffle/django-waffle#168]
We've been using Waffle for some time without issue, but recently upgraded to django 1.8 and waffle 0.10.1, and have started seeing a flood of these errors:

PicklingError: Can't pickle <class 'waffle.models.Flag_users'>: attribute lookup waffle.models.Flag_users failed

**Comment by Kobold** [comment]
+1, I'm seeing this as well. Restarting the server seems to fix it temporarily. Here's a stack trace:

```
Traceback (most recent call last):
  File "/.../local/lib/python2.7/site-packages/django/core/handlers/base.py", line 108, in get_response
    response = middleware_method(request)
  File "./findoffice/middleware.py", line 9, in process_request
    if waffle.flag_is_active(request, 'multiple_markets'):
  File "/.../local/lib/python2.7/site-packages/waffle/__init__.py", line 41, in flag_is_active
    cache_flag(instance=flag)
  File "/.../local/lib/python2.7/site-packages/waffle/models.py", line 119, in cache_flag
    cache.add(keyfmt(settings.FLAG_USERS_CACHE_KEY, f.name), f.users.all())
  File "/.../local/lib/python2.7/site-packages/django/core/cache/backends/locmem.py", line 38, in add
    pickled = pickle.dumps(value, pickle.HIGHEST_PROTOCOL)
PicklingError: Can't pickle <class 'waffle.models.Flag_users'>: attribute lookup waffle.models.Flag_users failed
```

Django==1.8.2
django-waffle==0.10.1

**Comment by Kobold** [comment]
It also seems that this is occurring only on flags that have users set on them.

**Comment by TechNickAI** [comment]
We're experiencing the same issue

Django==1.8.1
django-waffle==0.10.1

**Comment by TechNickAI** [comment]
As a workaround, I disabled caching for django waffle by setting `WAFFLE_CACHE_NAME` to a dummy cache backend.

Note: this will cause a lot of extra sql statements to your db. We use django-cacheops to cache at the sql level, so this layer of caching was redundant for us.

**Comment by ryanisnan** [comment]
We're doing the same workaround @gorillamania. Anyone had further experience with this?

**Comment by hakjoon** [comment]
I believe this was related to this bug with SimpleLazyObject which was fixed in 1.8.5 https://code.djangoproject.com/ticket/25389

I have not had this problem crop back up since upgrading to Django>=1.8.5

**Comment by jsocol** [comment]
Seems like this was a Django bug, but it's also going to be fixed by #211, which changes from caching whole user/group querysets to caching lists of IDs.
